text/3371-cargo-target-dir-templates.md

```diff
@@ -0,0 +1,411 @@
+-->
+
+- Do we want to differentiate according to users ? `bazel` is a generic build tool, whereas `cargo` is not, so maybe differentiating on users is not necessary for the latter ?
+
+## Rename the template from `{manifest-path-hash}` to `{project-hash}`
+
+This works in concert with the subsection `Introducing more data to the hash` later below: the currently proposed key name introduces a dependency in its name, should we prepare for possible changes by renaming it ?
+
+# Future possibilities
+[future-possibilities]: #future-possibilities
+
+<!--
+Think about what the natural extension and evolution of your proposal would
+be and how it would affect the language and project as a whole in a holistic
+way. Try to use this section as a tool to more fully consider all possible
+interactions with the project and language in your proposal.
+Also consider how this all fits into the roadmap for the project
+and of the relevant sub-team.
+
+This is also a good place to "dump ideas", if they are out of scope for the
+RFC you are writing but otherwise related.
+
+If you have tried and cannot think of any future possibilities,
+you may simply state that you cannot think of anything.
+
+Note that having something written down in the future-possibilities section
+is not a reason to accept the current or a future RFC; such notes should be
+in the section on motivation or rationale in this or subsequent RFCs.
+The section merely provides additional information.
+-->
+
+- Introduce remapping into the concept in some way.
+- Introduce a form of garbage collection for these target directories so we don't leak them when projects are deleted, see [rust-lang/cargo#13136](https://github.com/rust-lang/cargo#13136) (essentially, backlinks)
+
+## Adding `{xdg-cache-home}`, `{home}`, `{cargo-home}` or `{user-base-target-directory}` as templates
+
+OS-native cache directories are discussed more in details in [rust-lang/cargo#1734](https://github.com/rust-lang/cargo/issues/1734), there are semantic and naming issues to resolve before moving forward with them in cargo (and so as template for `CARGO_TARGET_DIR`).
+
+As a workaround, it is possible to use `CARGO_TARGET_DIR="${XDG_CACHE_HOME:-~/.cache}/cargo-target-directories/{manifest-path-hash}"`.
+
+It won't work in the `config.toml` but it will work with the environment variable and the command line option, both of which override the TOML config.
+
+It is certainly possible to add at least `{home}` and `{cargo-home}` but it can be done in the future without interfering at all with `{manifest-path-hash}`, making it a good option for a future addition without blocking.
+
+## Use templated `CARGO_TARGET_DIR` as the default instead of `target`
+
+1. This has several unresolved issues, notably: what default value do we use ? This discussion is already happening in [rust-lang/cargo#1734](https://github.com/rust-lang/cargo/issues/1734)) and does not block this RFC
```

Comment:
We have a default, what cargo script uses. 

Like I said, this is more of a foot note that moving caches out of `~/.cargo` is a foot note, rather than a concern.

Reply:
Done in https://github.com/rust-lang/rfcs/commit/d8797514d8b167c09f941b20606a409c35b6db3c